clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -495,8 +497,22 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new OperationNotAttemptedException(message);
             }
-        });
-
+    }),
+    KAFKA_STORAGE_ERROR(56, "Disk error when trying to access log file on the disk.",
+        new ApiExceptionBuilder() {
+            @Override
+            public ApiException build(String message) {
+                return new KafkaStorageException(message);
+            }
+    }),
+    UNKNOWN_ERROR_CODE(57, "The client received an unknown error code when processing the response. This may happen if " +
+            "the client library version is lower than the server library version",
+        new ApiExceptionBuilder() {
```

Comment:
The issue with retrying on any unknown error is that an app (e.g. MirrorMaker) could set the retry to infinite and then it won't be aware of the error.

Adding a new if/else statement for every new error code is not necessary bad as long as the process is clear. Adding UnknownErrorCodeException seems more complicated to me at this moment. Perhaps we could remove it and reconsider it later if there is a need.